kafka/consumer/subscription_state.py

```diff
@@ -120,6 +124,32 @@ def subscribe(self, topics=(), pattern=None, listener=None):
             raise TypeError('listener must be a ConsumerRebalanceListener')
         self.listener = listener
 
+    def _ensure_valid_topic_name(self, topic):
+        """ Ensures that the topic name is valid according to the kafka source.
+
+        Raises:
+            TypeError: if the topic is
+                        - a non-str or
+                        - an empty string or
+                        - the topic name is '.' or '..' or
+                        - the topic name does not consist of ASCII-characters/'-'/'_'/'.'
+        """
+
+        # See Kafka Source:
+        # https://github.com/apache/kafka/blob/39eb31feaeebfb184d98cc5d94da9148c2319d81/clients/src/main/java/org/apache/kafka/common/internals/Topic.java
+        if topic is None:
+            raise TypeError('All topics must not be None')
+        if not isinstance(topic, six.string_types):
+            raise TypeError('All topics must be strings')
+        if len(topic) == 0:
+            raise TypeError('All topics must be non-empty strings')
```

Comment:
I'd probably make the bottom 4 exceptions ValueError. (The first two make sense w/ TypeError)

Reply:
Done